internal/provider/resource_accept_risk.go

```diff
@@ -75,10 +75,11 @@ func ResourceAcceptRisk() *schema.Resource {
 				Default:     "any",
 			},
 			"protocol": {
-				Type:        schema.TypeString,
-				Description: descriptionProtocol,
-				Optional:    true,
-				Default:     "any",
+				Type:             schema.TypeString,
+				Description:      descriptionProtocol,
+				Optional:         true,
+				Default:          "any",
+				ValidateDiagFunc: validateRecastAcceptRiskProtocol,
```

Comment:
Validation should prevent improper writes, and your create/update logic looks to make sure what's written to the API will match what's retrieved.

A DiffSuppression function is needed though to prevent the fact that the terraform schema content coming from the user says 'TCP' from producing a diff when matched with the upstream read's  '4'.

Reply:
Look at the stuff in common.go https://github.com/palantir/terraform-provider-tenablesc/blob/develop/internal/provider/common.go#L127 for how diff suppression can work here - probably just take the newvalue and map it to the numeric val; if that matches then return true.